python/xir/parser.py

```diff
@@ -75,7 +75,7 @@ def program(self, args):
 
     def include(self, file_name):
         """Includ statements for external files"""
-        self._program._include.append(file_name[0])
+        self._program.add_include(file_name[0])
```

Comment:
Not much room to comment, but there's a `self._program._options = {k: v for k, v in args}` below which can be updated to use `self._program.add_option`.

Reply:
Ah, excellent catch. Thanks!